pkg/route/apis/route/v1/conversion.go

```diff
@@ -2,13 +2,43 @@ package v1
 
 import (
 	"k8s.io/apimachinery/pkg/runtime"
-
-	"github.com/openshift/origin/pkg/api/apihelpers"
-	routeapi "github.com/openshift/origin/pkg/route/apis/route"
 )
 
-func addConversionFuncs(scheme *runtime.Scheme) error {
-	return scheme.AddFieldLabelConversionFunc("v1", "Route",
-		apihelpers.GetFieldLabelConversionFunc(routeapi.RouteToSelectableFields(&routeapi.Route{}), nil),
-	)
+func addLegacyFieldConversionFuncs(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(LegacySchemeGroupVersion.String(), "Route", legacyRouteFieldSelectorConversionFunc); err != nil {
+		return err
+	}
+	return nil
+}
+
+func addFieldConversionFuncs(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.String(), "Route", routeFieldSelectorConversionFunc); err != nil {
+		return err
+	}
+	return nil
+}
+
+// because field selectors can vary in support by version they are exposed under, we have one function for each
+// groupVersion we're registering for
+
+func legacyRouteFieldSelectorConversionFunc(label, value string) (internalLabel, internalValue string, err error) {
+	switch label {
+	case "spec.path",
+		"spec.host",
+		"spec.to.name":
+		return label, value, nil
+	default:
+		return runtime.DefaultMetaV1FieldSelectorConversion(label, value)
```

Comment:
Looking at the other pr adding this to the rest, this struck me. Shouldn't that be `return apihelpers.LegacyMetaV1FieldSelectorConversionWithName(label, value)` ?

Reply:
> Looking at the other pr adding this to the rest, this struck me. Shouldn't that be return apihelpers.LegacyMetaV1FieldSelectorConversionWithName(label, value) ?

Interestingly enough, no.  The "name" -> "metadata.name" mapping never existed for routes.

Reply:
I see.